src/query.c

```diff
@@ -1753,63 +1780,92 @@ int QueryNode_ForEach(QueryNode *q, QueryNode_ForEachCallback callback, void *ct
+      STR_EQCASE(attr->name, attr->namelen, VECSIM_HYBRID_POLICY) ||
+      STR_EQCASE(attr->name, attr->namelen, VECSIM_BATCH_SIZE)) {
+    VecSimRawParam param = (VecSimRawParam){ .name = rm_strndup(attr->name, attr->namelen),
+                                            .nameLen = attr->namelen,
+                                            .value = rm_strndup(attr->value, attr->vallen),
+                                            .valLen = attr->vallen };
+    vq->params.params = array_ensure_append_1(vq->params.params, param);
+    bool resolve_required = false;  // at this point, we have the actual value in hand, not the query param.
+    vq->params.needResolve = array_ensure_append_1(vq->params.needResolve, resolve_required);
+    return 1;
+  }
+  return 0;
+}
+
 static int QueryNode_ApplyAttribute(QueryNode *qn, QueryAttribute *attr, QueryError *status) {
 
 #define MK_INVALID_VALUE()                                                         \
   QueryError_SetErrorFmt(status, QUERY_ESYNTAX, "Invalid value (%.*s) for `%.*s`", \
                          (int)attr->vallen, attr->value, (int)attr->namelen, attr->name)
 
+  int res = 0;
   // Apply slop: [-1 ... INF]
   if (STR_EQCASE(attr->name, attr->namelen, "slop")) {
     long long n;
     if (!ParseInteger(attr->value, &n) || n < -1) {
       MK_INVALID_VALUE();
-      return 0;
+      return res;
     }
     qn->opts.maxSlop = n;
+    res = 1;
 
   } else if (STR_EQCASE(attr->name, attr->namelen, "inorder")) {
     // Apply inorder: true|false
     int b;
     if (!ParseBoolean(attr->value, &b)) {
       MK_INVALID_VALUE();
-      return 0;
+      return res;
     }
     qn->opts.inOrder = b;
     qn->opts.flags |= QueryNode_OverriddenInOrder;
+    res = 1;
 
   } else if (STR_EQCASE(attr->name, attr->namelen, "weight")) {
     // Apply weight: [0  ... INF]
     double d;
     if (!ParseDouble(attr->value, &d) || d < 0) {
       MK_INVALID_VALUE();
-      return 0;
+      return res;
     }
     qn->opts.weight = d;
+    res = 1;
 
   } else if (STR_EQCASE(attr->name, attr->namelen, "phonetic")) {
     // Apply phonetic: true|false
     int b;
     if (!ParseBoolean(attr->value, &b)) {
       MK_INVALID_VALUE();
-      return 0;
+      return res;
     }
     if (b) {
       qn->opts.phonetic = PHONETIC_ENABLED;  // means we specifically asked for phonetic matching
     } else {
       qn->opts.phonetic =
           PHONETIC_DISABLED;  // means we specifically asked no for phonetic matching
     }
+    res = 1;
     // qn->opts.noPhonetic = PHONETIC_DEFAULT -> means no special asks regarding phonetics
     //                                          will be enable if field was declared phonetic
 
-  } else {
-    QueryError_SetErrorFmt(status, QUERY_ENOOPTION, "Invalid attribute %.*s", (int)attr->namelen,
-                           attr->name);
-    return 0;
+  } else if (STR_EQCASE(attr->name, attr->namelen, YIELD_DISTANCE_ATTRIBUTE) && qn->opts.flags & QueryNode_YieldsDistance) {
```

Comment:
why do we need the flag?

Reply:
To know if the `$yield_distance_as` attribute is valid for the current node

Reply:
We should check later how to handle the yield function.
Ignore or return a `inf` or `ERR` in invalid.